packages/boot/src/booters/controller.booter.ts

```diff
@@ -0,0 +1,125 @@
+ */
+export class ControllerBooter implements Booter {
+  options: ControllerOptions;
+  projectRoot: string;
+  dirs: string[];
+  extensions: string[];
+  discovered: string[];
+
+  /**
+   *
+   * @param app Application instance
+   * @param bootConfig Config options for boot
+   */
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE) public app: Application,
+    @inject(BootBindings.BOOT_OPTIONS) public bootConfig: BootOptions,
+  ) {
+    if (!bootConfig.controllers) bootConfig.controllers = {};
+    this.options = bootConfig.controllers;
+    this.projectRoot = bootConfig.projectRoot;
+  }
+
+  /**
+   * This phase is responsible for configuring this Booter. It converts
+   * options and assigns default values for missing values so ther phases
+   * don't have to perform checks / conversions.
+   */
+  async configure() {
+    this.options = Object.assign({}, ControllerDefaults, this.options);
+
+    this.dirs = Array.isArray(this.options.dirs)
+      ? this.options.dirs
+      : [this.options.dirs];
+
+    this.extensions = Array.isArray(this.options.extensions)
+      ? this.options.extensions
+      : [this.options.extensions];
+  }
+
+  /**
+   * This phase is responsible for discovering artifact files based on the
+   * given parameters. Sets options.discovered to an array of discovered
+   * artifact files.
+   */
+  async discover() {
+    if (this.options.glob) {
+      this.discovered = await discoverFilesWithGlob(
+        this.options.glob,
+        this.projectRoot,
+      );
+    } else {
+      this.discovered = await discoverFiles(
+        this.dirs,
+        this.extensions,
+        this.options.nested,
+        this.projectRoot,
+      );
+    }
+  }
+
+  /**
+   * This phase is responsible for reading the discovered files, checking the
+   * Classes exported and binding them to the Application instance for use.
+   *
+   * It will skip any files it isn't able to load and throw an error containing
+   * a list of skipped files. Other files that were read will still be bound
+   * to the Application instance.
+   */
+  async load() {
+    const classes = await loadClassesFromFiles(this.discovered);
+    classes.forEach(cls => {
+      this.app.controller(cls);
+    });
+  }
+}
+
+/**
+ * Type definition for ControllerOptions. These are the options supported by
+ * this Booter.
+ *
+ * @param dirs String / String Array of directories to check for artifacts.
+ * Paths must be relative. Defaults to ['controllers']
+ * @param extensions String / String Array of file extensions to match artifact
+ * files in dirs. Defaults to ['.controller.js']
+ * @param nested Boolean to control if artifact discovery should check nested
+ * folders or not. Default to true
+ * @param discovered  An array of discovered files. This is set by the
+ * discover phase.
+ */
+export type ControllerOptions = {
+  dirs: string | string[];
+  extensions: string | string[];
+  nested: boolean;
+  glob?: string;
```

Comment:
No TSDoc for this property. Is it meant to be private? (Even if it is, it should be documented with `@private`)